File: patterns/01-design-patterns/02-structural/06-bridge/python/bridge.py
class PatternError(Exception):
    def __init__(self,code):super().__init__(code);self.code=code
class EmailChannel:
    def send(self,recipient,message):return f"email:{recipient}:{message}"
class SlackChannel:
    def send(self,recipient,message):return f"slack:{recipient}:{message}"
class Alert:
    prefix=""
    def __init__(self,channel):self.channel=channel
    def deliver(self,recipient,body):return self.channel.send(recipient,f"{self.prefix} {body}")
class IncidentAlert(Alert):prefix="[INCIDENT]"
class ReminderAlert(Alert):prefix="[REMINDER]"
def evaluate(input_data):
    deliveries=[]
    for item in input_data.get("notifications",[]):
        channel=EmailChannel() if item["channel"]=="email" else SlackChannel() if item["channel"]=="slack" else (_ for _ in ()).throw(PatternError("unsupported_channel"))
        alert=IncidentAlert(channel) if item["kind"]=="incident" else ReminderAlert(channel) if item["kind"]=="reminder" else (_ for _ in ()).throw(PatternError("unsupported_alert"))
        deliveries.append(alert.deliver(item["recipient"],item["body"]))
    return{"deliveries":deliveries}

File: patterns/01-design-patterns/02-structural/06-bridge/go/bridge.go
package bridge

type PatternError struct{ code string }

func (e *PatternError) Error() string { return e.code }
func (e *PatternError) Code() string  { return e.code }

type Channel interface{ Send(string, string) string }
type emailChannel struct{}

func (emailChannel) Send(recipient, message string) string {
	return "email:" + recipient + ":" + message
}

type slackChannel struct{}

func (slackChannel) Send(recipient, message string) string {
	return "slack:" + recipient + ":" + message
}

type Alert interface{ Deliver(string, string) string }
type alert struct {
	prefix  string
	channel Channel
}

func (a alert) Deliver(recipient, body string) string {
	return a.channel.Send(recipient, a.prefix+" "+body)
}
func Evaluate(input map[string]any) (any, error) {
	raw, _ := input["notifications"].([]any)
	deliveries := make([]string, 0, len(raw))
	for _, item := range raw {
		value := item.(map[string]any)
		var channel Channel
		switch value["channel"] {
		case "email":
			channel = emailChannel{}
		case "slack":
			channel = slackChannel{}
		default:
			return nil, &PatternError{code: "unsupported_channel"}
		}
		prefix := ""
		switch value["kind"] {
		case "incident":
			prefix = "[INCIDENT]"
		case "reminder":
			prefix = "[REMINDER]"
		default:
			return nil, &PatternError{code: "unsupported_alert"}
		}
		var current Alert = alert{prefix, channel}
		deliveries = append(deliveries, current.Deliver(value["recipient"].(string), value["body"].(string)))
	}
	return map[string]any{"deliveries": deliveries}, nil
}
